app/forms/SelectableFormProvider.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms
+
+import forms.mappings.Mappings
+import models.reference.Selectable
+import models.{CheckMode, Index, Mode, NormalMode, SelectableList}
+import play.api.data.Form
+import play.api.i18n.Messages
+
+import javax.inject.Inject
+
+class SelectableFormProvider @Inject() extends Mappings {
+
+  def apply[T <: Selectable](mode: Mode, houseConsignmentIndex: Index, itemIndex: Index, prefix: String, selectableList: SelectableList[T], args: Any*)(implicit
```

Comment:
I think this can be written without explicitly passing the indexes into the apply method, they can come in with the args.

Reply:
This way other pages that are not under the houseConsignment directory can use the form provider.

Reply:
Yeah that would be nice, let me see 👀